src/bun.js/webcore/encoding.zig

```diff
@@ -559,6 +559,13 @@ pub const TextDecoder = struct {
                     remainder = remainder[1..];
                     continue;
                 },
+                // BOM handling
+                0xFEFF => {
```

Comment:
This should only be in there for the first character right? This code runs on every character

Reply:
We handle the first character of BOM in `decodeSlice`.

In unicode `0xFEFF` is a special, we should not rewrite it to `0xFFFD` in else block, just keep it. This is to be consistent with node.